recipes/dev/lora_finetune_distributed_multi_dataset.py

```diff
@@ -828,13 +827,15 @@ def train(self) -> None:
                 with self.activations_handling_ctx:
                     outputs = self._model(**batch)
 
-                if self.linear_loss:
-                    weight = self._model.linear_projection_weight
-                    current_loss = self._loss_fn(weight, outputs, labels)
-                else:
+                # post process for third party loss functions
+                if not isinstance(self._loss_fn, modules.loss.SFTLoss):
                     labels = labels.reshape(-1)
                     outputs = outputs.reshape(-1, outputs.size(-1))
-                    current_loss = self._loss_fn(outputs, labels)
+                    if isinstance(outputs, DTensor):
```

Comment:
nit but would at least add a comment explaining when/why we need this

Reply:
I guess.. as a reader of that my next question would be "why is the tensor sharded to begin with?" In normal cases of distributed training it's not required to do this manually, right?